### ISSUE TITLE
MyISAM support is blocking use of Azure Database for MySQL

### ISSUE THREAD

**Issue by mwjames** [SemanticMediaWiki/SemanticMediaWiki#2661]
Picking up https://www.mediawiki.org/wiki/Topic:Tu4rane5czgppcjn

> attempted to move my mediawiki DB into Azure Database for MySql (their new hosted MySql offering). Got blocked on the use of MyISAM fulltext search on smw_ft_search.
> Can this be migrated to InnoDB soon? Latest MySQLs have support for fulltext in InnoDB.

Modifying the `LocalSettings.php` with `$smwgFulltextSearchTableOptions['mysql'] = [ 'ENGINE=InnoDB, DEFAULT CHARSET=utf8' ]` should resolve the dependency on `MyISAM`.

**Comment by kghbln** [comment]
[Documented on wiki](https://www.semantic-mediawiki.org/w/index.php?title=Help%3A%24smwgFulltextSearchTableOptions&type=revision&diff=54144&oldid=52062)

**Comment by emacarron** [comment]
In case this helps, I installed MediaWiki on Azure using the Hallo Welt! - Medienwerkstatt GmbH available in the market. 

These are the changes I did to make it work:

- Used MySql 5.7
- Changed tables.sql in this way [1]
- Used UTF-8 as colation during the installation

Go to D:\home\site\wwwroot\maintenance\tables.sql and leave this table creation this way:

CREATE TABLE /*_*/searchindex (
  -- Key to page_id
  si_page int unsigned NOT NULL,
  -- Munged version of title
  si_title varchar(255) NOT NULL default '',
  -- Munged version of body text
  si_text mediumtext NOT NULL
) /*$wgDBTableOptions*/;

-- former line was: ) ENGINE=MyISAM DEFAULT CHARSET=utf8;

So... looks like InnoDB can be fully supported by adding MySQL 5.7 (maybe 5.6) as a pre-requisite and forcing the installation to work on UTF-8.
